Applications/Spire/Source/Ui/OverlayPanel.cpp

```diff
@@ -139,6 +139,16 @@ bool OverlayPanel::event(QEvent* event) {
   return QWidget::event(event);
 }
 
+void OverlayPanel::keyPressEvent(QKeyEvent* event) {
+  if(event->key() == Qt::Key_Escape) {
+    if(!m_is_closed) {
+      fade(true);
+    }
```

Comment:
In cases like this I think we usually do an 'early return', where after the second if statement the method would return, then the call to `QWidget::keyPressEvent(event)` wouldn't have to be an else statement.